standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/RedshiftConnectorProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.apache.hadoop.hive.metastore.dataconnector.jdbc;
+
+import org.apache.hadoop.hive.metastore.ColumnType;
+import org.apache.hadoop.hive.metastore.api.DataConnector;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class RedshiftConnectorProvider extends AbstractJDBCConnectorProvider {
+    private static Logger LOG = LoggerFactory.getLogger(RedshiftConnectorProvider.class);
+
+    private static final String DRIVER_CLASS = "com.amazon.redshift.jdbc42.Driver".intern();
+
+    public RedshiftConnectorProvider(String dbName, DataConnector dataConn) {
+        super(dbName, dataConn, DRIVER_CLASS);
+    }
+
+    protected String getDataType(String dbDataType, int size) {
+        String mappedType = super.getDataType(dbDataType, size);
+
+        // The VOID type points to the corresponding datatype not existing in hive. These datatypes are datastore
+        // specific. They need special handling. An example would be the Geometric type that is not supported in Hive.
+        // The other cases where a datatype in redshift is resolved to a VOID type are during the use of aliases like
+        // float8, int8 etc. These can be mapped to existing hive types and are done below.
+        if (!mappedType.equalsIgnoreCase(ColumnType.VOID_TYPE_NAME)) {
+            return mappedType;
+        }
```

Comment:
```suggestion
```
The `if` is not necessary and possibly wrong. In terms of inheritance it is more natural to continue to the switch case below to attempt to override the `mappedType` if necessary. To do that we have to skip the `default` case below.